src/praisonai-train/praisonai_train/train/_ollama.py

```
@@ -5,93 +5,221 @@
+
     # Check if ollama CLI is available
     if shutil.which("ollama") is None:
         raise RuntimeError("`ollama` CLI not found; install from https://ollama.com")
-    
-    # Start daemon in detached mode
+
+    # Capture serve stderr to a temp file so a startup failure (e.g. port in use,
+    # bad OLLAMA_HOST) can be surfaced in the timeout message instead of vanishing.
+    serve_err = tempfile.TemporaryFile(mode="w+")
     proc = subprocess.Popen(
         ["ollama", "serve"],
```

Ruff (0.15.21)

[error] Starting a process with a partial executable path

(S607)